File: 199. Binary Tree Right Side View/rightSideView.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class rightSideView {
    public List<Integer> solution(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if (root == null) {
            return res;
        }
        Queue<TreeNode> q1 = new LinkedList<>();
        q1.offer(root);
        while (!q1.isEmpty()) {
            int size = q1.size();
            for (int i = 0; i < size; i ++) {
                TreeNode temp = q1.poll();
                if (temp.left != null) {
                    q1.offer(temp.left);
                }
                if (temp.right != null) {
                    q1.offer(temp.right);
                }
                if (i == size - 1) {
                    res.add(temp.val);
                }
            }
        }
        return res;
    }
}

File: 199. Binary Tree Right Side View/rightSideView.py
import queue
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None
class rightSideView:
    def solution(self, root: TreeNode) -> List[int]:
        if root == None:
            return []
        res = []
        q1 = queue.Queue()
        q1.put(root)
        while not q1.empty():
            size = q1.qsize()
            for i in range(size):
                temp = q1.get()
                if temp.left != None:
                    q1.put(temp.left)
                if temp.right != None:
                    q1.put(temp.right)
                if i == size - 1:
                    res.append(temp.val)
        return res
            
            
            
